soc/nordic/nrf54h/soc.c

```diff
@@ -87,6 +91,17 @@ static int trim_hsfll(void)
 	return 0;
 }
 
+#if defined(CONFIG_ARM_ON_ENTER_CPU_IDLE_HOOK)
+bool z_arm_on_enter_cpu_idle(void)
```

Comment:
Should the `_pre_sleep()` function be called for other sleep states than "Idle"? I'm thinking mostly about Suspend-to-RAM

Reply:
yes but s2ram (#73095) is not yet merged.

Reply:
Couldn't it just be called for any idle state not matter if s2ram is available or not?

Reply:
> Couldn't it just be called for any idle state not matter if s2ram is available or not?

I guess you mean "any sleep state". _Idle_ is a specific light sleep (runtime idle). Ref https://docs.zephyrproject.org/latest/doxygen/html/group__subsys__pm__states.html#ga20e2f5ea9027a3653e5b9cc5aa1e21d5

We don't have a generic hooks infrastructure for all sleep states. I guess it could be added to simplify things like the one we discussed.

Reply:
Yea sorry I mixed it up, I thought s2ram would also enter up in this `z_arm_on_enter_cpu_idle()` here and that's what I meant with 'any idle state'. I forgot `sys_suspend_to_ram` isn't calling `k_cpu_idle()` but straight `__WFI()` at the end

Reply:
Yes, the exact reason is that _suspend to idle_ or _runtime idle_ require other hooks to manage power domains than _suspend to RAM_.